FileManagerWpf/Services/FileManager.cs

```diff
@@ -0,0 +1,208 @@
+                                    | NotifyFilters.FileName
+                                    | NotifyFilters.LastAccess
+                                    | NotifyFilters.LastWrite
+                                    | NotifyFilters.Security
+                                    | NotifyFilters.Size;
+
+            _watcher.Changed += OnChanges;
+            _watcher.Created += OnChanges;
+            _watcher.Deleted += OnChanges;
+            _watcher.Renamed += OnChanges;
+            _onChanges = onChanges;          
+        }      
+
+        public IEnumerable<DriveItem> GetDrives()
+        {
+            foreach (var drive in DriveInfo.GetDrives())
+            {
+                yield return new DriveItem
+                {
+                    Path = drive.RootDirectory.FullName,
+                    Name = drive.RootDirectory.Name,
+                    Size = drive.TotalSize,
+                    TotalFreeSpace = drive.TotalFreeSpace,
+                    UsedSpace = drive.TotalSize - drive.TotalFreeSpace
+                };
+            }
+        }
+        public IEnumerable<FolderItem> GetSubDirectories(TabViewModel tabViewModel, string path)
+        {
+            if (!Directory.Exists(path)) throw new DirectoryNotFoundException();
+
+            foreach (var dir in Directory.GetDirectories(path))
+            {
+                var di = new DirectoryInfo(dir);
+                yield return new FolderItem
+                {
+                    TabViewModel = tabViewModel,
+                    Path = dir,
+                    Size = 0,
+                    Name = di.Name,
+                    Type = ItemType.Dir,
+                    Date = di.CreationTime.ToString(CultureInfo.CurrentCulture)
+                };
+            }
+        }
+        public IEnumerable<FileItem> GetFiles(TabViewModel tabViewModel, string path)
+        {
+            if (!Directory.Exists(path)) throw new DirectoryNotFoundException();
+
+            foreach (var f in Directory.GetFiles(path))
+            {
+                var fi = new FileInfo(f);
+                yield return new FileItem
+                {
+                    TabViewModel = tabViewModel,
+                    Path = fi.FullName,
+                    Size = fi.Length,
+                    Name = Path.GetFileNameWithoutExtension(fi.FullName),
+                    NameWithExt = fi.Name,
+                    Ext = fi.Extension.Replace(".",""),
+                    Type = ItemType.File,
+                    Date = fi.CreationTime.ToString(CultureInfo.CurrentCulture)
+                };
+            }
+        }
+        public void CreateNewFolder(string path, string name)
+        {
+            var pathNewFolder = Path.Combine(path, name);
+            Directory.CreateDirectory(pathNewFolder);
+        }
+        public void CreateNewFile(string path, string name)
+        {
+            var pathNewFile = Path.Combine(path, name);
+            File.Create(pathNewFile);
```

Comment:
```suggestion
            File.Create(pathNewFile).Dispose();
```
Возвращается открытый поток. Надо закрыть.